pkg/artifactory/provider.go

```diff
@@ -224,19 +224,32 @@ func providerConfigure(d *schema.ResourceData, terraformVersion string) (interfa
 func checkArtifactoryLicense(client *resty.Client) error {
 
 	type License struct {
-		Type         string `json:"type"`
-		ValidThrough string `json:"validThrough"`
-		LicensedTo   string `json:"licensedTo"`
+		Type string `json:"type"`
+	}
+
+	type LicensesWrapper struct {
+		License
+		Licenses []License `json:"licenses"` // HA licenses returns as an array instead
 	}
 
-	license := License{}
-	_, err := client.R().SetResult(&license).Get("/artifactory/api/system/licenses/")
+	licensesWrapper := LicensesWrapper{}
+	_, err := client.R().
+		SetResult(&licensesWrapper).
+		Get("/artifactory/api/system/licenses")
+
 	if err != nil {
-		return err
+		return fmt.Errorf("Failed to check for license. If your usage doesn't require admin permission, you can set `check_license` attribute to `false` to skip this check. %s", err)
+	}
+
+	var licenseType string
+	if len(licensesWrapper.Licenses) > 0 {
+		licenseType = licensesWrapper.Licenses[0].Type
```

Comment:
Why index 0

Reply:
The HA license is a collection of same license text with varying node ID, etc. But the `type` doesn't change. So checking the first one is sufficient.

Reply:
Are you certain it will never be different? I mean, this is RT we are talking about and I'd put nothing past them.

On second thought, I don't know what the hell to do if these license types came back mixed